File: android/app/src/main/java/com/shadelight/iausage/data/ProviderVisuals.kt
package com.shadelight.iausage.data

import com.shadelight.iausage.R

/** Same brand icon set as Desktop/VSCode, converted from the canonical SVGs
 * under src/assets/providers into vector drawables (see
 * res/drawable/ic_provider_*.xml). A provider without a shipped icon never
 * breaks the UI: it falls back to a generic mark + a derived 3-letter code. */
data class ProviderVisual(val icon: Int?, val shortCode: String)

private val KNOWN_ICONS: Map<String, Int> = mapOf(
    "anthropic" to R.drawable.ic_provider_anthropic,
    "openai" to R.drawable.ic_provider_openai,
    "cursor" to R.drawable.ic_provider_cursor,
    "antigravity" to R.drawable.ic_provider_antigravity,
    "opencode_go" to R.drawable.ic_provider_opencode_go,
    "openrouter" to R.drawable.ic_provider_openrouter,
    "deepseek" to R.drawable.ic_provider_deepseek,
    "groq" to R.drawable.ic_provider_groq,
    "kimi" to R.drawable.ic_provider_kimi,
    "kilo" to R.drawable.ic_provider_kilo,
    "minimax" to R.drawable.ic_provider_minimax,
    "grok" to R.drawable.ic_provider_grok,
    "copilot" to R.drawable.ic_provider_copilot,
    "windsurf" to R.drawable.ic_provider_windsurf,
    "zai" to R.drawable.ic_provider_zai,
    "moonshot" to R.drawable.ic_provider_moonshot,
    "novita" to R.drawable.ic_provider_novita,
    "kiro" to R.drawable.ic_provider_kiro,
    "nous" to R.drawable.ic_provider_nous,
)

private fun fallbackShortCode(name: String): String {
    val letters = name.filter { it.isLetterOrDigit() }
    return (if (letters.length >= 3) letters.substring(0, 3) else letters.ifEmpty { "AI" }).uppercase()
}

fun providerVisual(id: String, name: String): ProviderVisual =
    ProviderVisual(icon = KNOWN_ICONS[id], shortCode = fallbackShortCode(name))

File: apps/vscode-extension/src/status/provider-visuals.ts
export interface ProviderVisual {
  /** Codicon/Product Icon id, usable as `$(id)` in status bar text or Markdown. */
  icon: string;
  /** 3-letter compact code shown in "compact" display mode. */
  short: string;
}

// Registered in package.json's contributes.icons, backed by
// media/provider-icons/provider-icons.woff (built from the same brand SVGs
// the desktop app ships under src/assets/providers, via `npm run build-icons`).
const KNOWN_VISUALS: Record<string, ProviderVisual> = {
  anthropic: { icon: "ia-anthropic", short: "CLD" },
  openai: { icon: "ia-openai", short: "CDX" },
  cursor: { icon: "ia-cursor", short: "CUR" },
  antigravity: { icon: "ia-antigravity", short: "AGY" },
  opencode_go: { icon: "ia-opencode_go", short: "OCG" },
  openrouter: { icon: "ia-openrouter", short: "ORT" },
  deepseek: { icon: "ia-deepseek", short: "DSK" },
  groq: { icon: "ia-groq", short: "GRQ" },
  kimi: { icon: "ia-kimi", short: "KIM" },
  kilo: { icon: "ia-kilo", short: "KLO" },
  minimax: { icon: "ia-minimax", short: "MMX" },
  grok: { icon: "ia-grok", short: "GRK" },
  copilot: { icon: "ia-copilot", short: "CPT" },
  windsurf: { icon: "ia-windsurf", short: "WSF" },
  zai: { icon: "ia-zai", short: "ZAI" },
  moonshot: { icon: "ia-moonshot", short: "MSH" },
  novita: { icon: "ia-novita", short: "NOV" },
  kiro: { icon: "ia-kiro", short: "KIR" },
  nous: { icon: "ia-nous", short: "NOU" },
};

function fallbackShortCode(name: string): string {
  const letters = name.replace(/[^A-Za-z0-9]/g, "");
  return (letters.slice(0, 3) || "AI").toUpperCase();
}

/** A provider IA Usage doesn't ship a real logo for yet never breaks the
 * status bar: it falls back to a generic dot icon and a derived 3-letter code. */
export function providerVisual(id: string, name: string): ProviderVisual {
  return KNOWN_VISUALS[id] ?? { icon: "circle-filled", short: fallbackShortCode(name) };
}
